src/components/anchor-navigation/anchor-navigation.component.js

```diff
@@ -0,0 +1,133 @@
+import React, {
+  useState, useRef, useEffect, useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+
+import Event from '../../utils/helpers/events';
+import { StyledAnchorNavigation, StyledNavigation, StyledContent } from './anchor-navigation.style';
+import AnchorNavigationItem from './anchor-navigation-item.component';
+
+const AnchorNavigation = ({ children, stickyNavigation, styleOverride }) => {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  const sectionRefs = useRef(React.Children.map(stickyNavigation.props.children, child => child.props.target));
+
+  const anchorRefs = useRef(Array.from({
+    length: React.Children.count(stickyNavigation.props.children)
+  }, () => React.createRef()));
+
+  const contentRef = useRef();
+
+  const navigationRef = useRef();
+
+  const setSelectedAnchorBasedOnScroll = useCallback(throttle(() => {
+    const topEdgeOffsets = sectionRefs.current.map(({ current }) => current.getBoundingClientRect().top);
+    const { top: navTopOffset } = navigationRef.current.getBoundingClientRect();
+    const indexOfSmallestNegativeTopOffset = topEdgeOffsets.reduce((currentTopIndex, topOffset, index) => {
+      if (topOffset > navTopOffset) return currentTopIndex;
+      return topOffset > topEdgeOffsets[currentTopIndex] ? index : currentTopIndex;
+    }, 0);
+
+    setSelectedIndex(indexOfSmallestNegativeTopOffset);
+  }, 200), []);
+
+  useEffect(() => {
+    window.addEventListener('scroll', setSelectedAnchorBasedOnScroll, true);
+    return () => window.removeEventListener('scroll', setSelectedAnchorBasedOnScroll, true);
+  }, [setSelectedAnchorBasedOnScroll]);
+
+  const scrollToItem = (event, index) => {
+    event.preventDefault();
+    const sectionToScroll = sectionRefs.current[index].current;
+    sectionToScroll.scrollIntoView({ block: 'start', inline: 'nearest', behavior: 'smooth' });
```

Comment:
Passing `scrollIntoView` the `scrollIntoViewOptions` isn't supported in `IE` or `Safari`, is the degradation in behaviour acceptable or do we need some kind of alternative for those browsers here?
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

Reply:
I thought only `IE11` doesn't support it, didn't even check for Safari. I think that getting immediately scrolled to the wanted section is fine for `IE11` as it covers bare minimum of the functionality. But for Safari it might be not enough. I've looked for quick solutions and it seems that installing a polyfill would do the job.
https://github.com/iamdustan/smoothscroll - would work, but I'm not sure what is our stance on installing additional packages like this

Reply:
@harpalsingh Confirmed that it's ok that on Safari it will just immediately jump to the wanted section. So I'm leaving it as it is.

Reply:
Thanks for clearing that up @mkrds

Reply:
Sorry, was meant to comment that on the ticket, yes that was ok about Safari.